### PR TITLE
Remove TestPyPI tokens from wheel recipes

### PR DESCRIPTION
**Pull request by maliasadi**
### Before submitting

Please complete the following checklist when submitting a PR:

- [ ] All new features must include a unit test.
      If you've fixed a bug or added code that should be tested, add a test to the
      [`tests`](../tests) directory!

- [ ] All new functions and code must be clearly commented and documented.
      If you do make documentation changes, make sure that the docs build and
      render correctly by running `make docs`.

- [ ] Ensure that the test suite passes, by running `make test`.

- [x] Add a new entry to the `.github/CHANGELOG.md` file, summarizing the
      change, and including a link back to the PR.

- [x] Ensure that code is properly formatted by running `make format`. 

When all the above are checked, delete everything above the dashed
line and fill in the pull request template.

------------------------------------------------------------------------------------------------------------

**Context:**
Currently, Linux and macOS (x86_64 & AARCH64) wheels are failing due to the use of older legacy tokens during the upload to TestPyPI. This PR fixes the issue by registering the wheel recipes as Trusted Publishing and increases the verbosity of this action step to provide a detailed description for future failures.  

**Description of the Change:**

**Benefits:**
No need to use secret tokens as these actions are now registered as TP at https://test.pypi.org/

**Possible Drawbacks:**

**Related GitHub Issues:**

[sc-77894]

### CODE REVIEW

File: .github/workflows/wheel_linux_x86_64.yml [hunk 1]
```diff
@@ -209,19 +209,20 @@ jobs:
...
         cibw_build: ${{ fromJson(needs.set_wheel_build_matrix.outputs.python_version) }}
     runs-on: ubuntu-latest
-    if: |
-      github.event_name == 'release' ||
-      github.ref == 'refs/heads/master'
+    permissions:
+      id-token: write
+    # if: |
+    #   github.event_name == 'release' ||
+    #   github.ref == 'refs/heads/master'
```
Comment (maliasadi) [inline .github/workflows/wheel_linux_x86_64.yml:216]:
```suggestion
    if: |
      github.event_name == 'release' ||
      github.ref == 'refs/heads/master'
```

Comment (maliasadi) [reply .github/workflows/wheel_linux_x86_64.yml:216]: I'll remove it after re-triggering the wheels..

File: .github/workflows/wheel_macos_arm64.yml [hunk 2]
```diff
@@ -143,19 +143,20 @@ jobs:
...
         cibw_build: ${{ fromJson(needs.mac-set-matrix-arm.outputs.python_version) }}
     runs-on: ubuntu-latest
-    if: |
-      github.event_name == 'release' ||
-      github.ref == 'refs/heads/master'
+    permissions:
+      id-token: write
+    # if: |
+    #   github.event_name == 'release' ||
+    #   github.ref == 'refs/heads/master'
```
Comment (maliasadi) [inline .github/workflows/wheel_macos_arm64.yml:150]: Same here.

**Review by AmintorDusko — COMMENTED** [review]
Could you please provide more context for this PR?
Perhaps filling the PR description.

**Comment by maliasadi** [comment]
@AmintorDusko Didn't expect your review to be this fast :smile:

**Review by LuisAlfredoNu — APPROVED** [review]
LGTM @maliasadi 
Thank you!!! :rocket:

**Pull request merged**